### PR TITLE
Minor Updates

### PR DESCRIPTION
**Pull request by zhollis21**
Feel free to look at the whole file

### CODE REVIEW

**Comment by zhollis21** [comment]
image

File: TaxCalculator/TaxCalculator.cs [hunk 1]
```diff
@@ -47,29 +47,108 @@ public void GetFinancialInformationFromUser()
...
+            var taxBrackets = _isMarried ? _married2019TaxBrackets : _single2019TaxBrackets;
+
+            long previousBracketIncomeLimit = 0;
+            foreach (var bracket in taxBrackets)
+            {
+                long bracketRange = bracket.IncomeLimit - previousBracketIncomeLimit;
+
+                double applicableUntaxedIncome;
+
+                // How much income are we taxing in this bracket?
```
Comment (paulaOliverosHollis) [inline TaxCalculator/TaxCalculator.cs]: The first two comments are not very clear. It is really hard for me to know what is going by just looking at the comments. Maybe come up with a different way of saying what you are trying to say. It seems to me that those comments sound a lot like what YOU personally were thinking at the moment; almost like following your thought process. I suggest you come up with comments that describe the type of calculation you are executing in that particular part. Keep it action-oriented.

Comment (zhollis21) [reply TaxCalculator/TaxCalculator.cs]: Ok, good feedback.

Give me suggestions of what you would put for the comments...

**Comment by zhollis21** [comment]
> Good job fixing this bug right away!

Thanks!

File: TaxCalculator/TaxCalculator.cs [hunk 2]
```diff
@@ -82,9 +161,9 @@ private static bool GetYesOrNoAnswer(string prompt)
             return answer == 'y';
```
Comment (paulaOliverosHollis) [inline TaxCalculator/TaxCalculator.cs:164]: This is a clever way of handling validation. Great job!

**Review by paulaOliverosHollis — CHANGES_REQUESTED** [review]
Great Job! I really like the structure you kept throughout the program. You did a good job making sure that all the actions and calculations your program is doing, are very well distributed and separated into their own functions. You need a little bit more work on the comments you already have and more comments in the last function where you are making most of the calculations. It is a very long function so I think that it would help to have a few comments here and there.

File: TaxCalculator/TaxCalculator.cs [hunk 3]
```diff
@@ -47,29 +47,109 @@ public void GetFinancialInformationFromUser()
...
+        public void CalculateTaxEstimation()
+        {
+            double untaxedIncome = (_grossIncome + _spouseGrossIncome) - _taxExemptIncome;
+            long totalIncome = _grossIncome + _spouseGrossIncome;
+            double taxEstimate = 0;
+
+            // Select the correct tax bracket based on marriage status
+            var taxBrackets = _isMarried ? _married2019TaxBrackets : _single2019TaxBrackets;
+
+            // Used to determine the range of a tax bracket by subracting
```
Comment (paulaOliverosHollis) [inline TaxCalculator/TaxCalculator.cs]:
```suggestion
            // Used to determine the range of a tax bracket by subtracting 

```

File: TaxCalculator/TaxCalculator.cs [hunk 4]
```diff
@@ -47,29 +47,109 @@ public void GetFinancialInformationFromUser()
...
+            // the current income limit from the previous income limit.
+            long previousBracketIncomeLimit = 0;
+
+            foreach (var bracket in taxBrackets)
+            {
+                // Calculate the range of income that can be taxed under this bracket
+                long bracketRange = bracket.IncomeLimit - previousBracketIncomeLimit;
+                
+                // If the user has less untaxed income than the bracket range, 
+                // then we can only tax how much untaxed income they have left
```
Comment (paulaOliverosHollis) [inline TaxCalculator/TaxCalculator.cs]:
```suggestion
                // then we can only tax however much untaxed income they have left

```

File: TaxCalculator/TaxCalculator.cs [hunk 5]
```diff
@@ -47,29 +47,109 @@ public void GetFinancialInformationFromUser()
...
+
         private void IdentifyExemptIncome()
         {
             _taxExemptIncome = 0;
 
             bool has401k = GetYesOrNoAnswer("Do you have a non-Roth 401k?");
             if (has401k)
             {
-                _taxExemptIncome += (GetNonNegativePercent("Enter the contribution percentage") / 100) * _grossIncome;
+                var percentContribution = GetNonNegativePercent("Enter the contribution percentage");
```
Comment (paulaOliverosHollis) [inline TaxCalculator/TaxCalculator.cs:105]: Great variable names.

File: TaxCalculator/TaxCalculator.cs [hunk 6]
```diff
@@ -47,29 +47,109 @@ public void GetFinancialInformationFromUser()
...
+            if (_isMarried)
+            {
+                _spouseGrossIncome += GetNonNegativeLong("Enter your spouse's yearly income");
+            }
+
+            Console.WriteLine();
         }
 
-        private static bool GetYesOrNoAnswer(string prompt)
+        private static bool GetYesOrNoAnswer(string question)
```
Comment (paulaOliverosHollis) [inline TaxCalculator/TaxCalculator.cs:152]: I like that you are using "question" instead of "prompt".

**Review by paulaOliverosHollis — DISMISSED** [review]
Great job! The comments are very helpful and the variable names are descriptive and easy to follow. Very clean and nifty program!

**Review by paulaOliverosHollis — APPROVED** [review]
Great job!

**Pull request merged**